moveit_ros/moveit_servo/include/moveit_servo/utilities.h

```diff
@@ -72,4 +73,18 @@ double velocityScalingFactorForSingularity(const moveit::core::JointModelGroup*
                                            const double leaving_singularity_threshold_multiplier, rclcpp::Clock& clock,
                                            const moveit::core::RobotStatePtr& current_state, StatusCode& status);
 
+/** \brief Joint-wise update of a sensor_msgs::msg::JointState with given delta's
+ * Also filters and calculates the previous velocity
+ * @param clock A ROS clock, for logging
+ * @param publish_period The publishing rate for servo command
+ * @param delta_theta Eigen vector of joint delta's
+ * @param previous_joint_state The previous joint state
+ * @param next_joint_state The joint state object which will hold the next joint state.
+ * @param smoother The trajectory smoother to be used.
+ * @return Returns false if there is a problem, true otherwise
+ */
+bool applyJointUpdate(rclcpp::Clock& clock, const double publish_period, const Eigen::ArrayXd& delta_theta,
+                      const sensor_msgs::msg::JointState& previous_joint_state,
+                      sensor_msgs::msg::JointState& next_joint_state,
+                      std::shared_ptr<online_signal_smoothing::SmoothingBaseClass>& smoother);
```

Comment:
That sounds reasonable.
In that case we would also need to replace [this](https://github.com/ros-planning/moveit2/blob/d343ed5b793c4bdb207b9df02200d60c9c4ccc68/moveit_ros/moveit_servo/src/servo_calcs.cpp#L176) , with `createUniqueInstance()` instead?
Since this is only used within `ServoCalcs` and only one copy is needed.

Reply:
Used a `pluginlib::UniquePtr` here since that is what `createUniqueInstance()` returns.